### PR TITLE
chore: sort the request headers output by the _M.uri and _M.old_uri  function in t/lib/server.lua

### LINKED ISSUE #2636: chore: sort the request headers output by the _M.uri and _M.old_uri function in 't/lib/server.lua'
### Issue description

In writing test cases, when the `_M.uri()` method in `t/lib/server.lua` is used, the order of output request headers is unstable, resulting in test cases sometimes failing. We need to sort the headers obtained by `local headers = ngx.req.get_headers()` to make it stable.

Need to modify the code: https://github.com/apache/apisix/blob/master/t/lib/server.lua#L103-L106
Modification method reference:: https://github.com/apache/apisix/blob/master/t/plugin/proxy-mirror.t#L40-L49

### PR DESCRIPTION
**Pull request by Firstsawyou**
fix #2636

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: t/lib/server.lua [hunk 1]
```diff
@@ -101,8 +101,15 @@ function _M.uri()
     -- ngx.sleep(1)
     ngx.say("uri: ", ngx.var.uri)
     local headers = ngx.req.get_headers()
-    for k, v in pairs(headers) do
-        ngx.say(k, ": ", v)
+
+    local keys = {}
+    for k, _ in pairs(headers) do
```
Comment (tokers) [inline t/lib/server.lua:106]: Just ignore the second dumb variable if you don't need it.

```lua
for k in pairs(headers) do
      ......
end
```

File: t/lib/server.lua [hunk 2]
```diff
@@ -101,8 +101,15 @@ function _M.uri()
...
+        table.insert(keys, k)
+    end
+    table.sort(keys)
+
+    for _, key in pairs(keys) do
```
Comment (tokers) [inline t/lib/server.lua:111]: Use `ipairs` here since `keys` is an array-like table.

File: t/lib/server.lua (hunk 1 above)
Comment (Firstsawyou) [reply t/lib/server.lua:106]: nice.

Comment (Firstsawyou) [reply t/lib/server.lua]: fixed.

File: t/lib/server.lua (hunk 2 above)
Comment (Firstsawyou) [reply t/lib/server.lua]: fixed.

File: t/plugin/hmac-auth.t [hunk 3]
```diff
@@ -1246,16 +1246,16 @@ location /t {
 GET /t
 --- response_body
 uri: /uri
-host: 127.0.0.1
+content-length: 52
 content-type: application/x-www-form-urlencoded
-x-real-ip: 127.0.0.1
+host: 127.0.0.1
 user-agent: lua-resty-http/0.14 (Lua) ngx_lua/10017
```
Comment (nic-chen) [inline t/plugin/hmac-auth.t:1252]: a little confuse, why `user-agent` is in this place ?

Comment (Firstsawyou) [reply t/plugin/hmac-auth.t:1252]: When requesting the URL of `/uri`, all request headers will be output. So it is normal to have the request header `user-agent`.

Comment (spacewander) [reply t/plugin/hmac-auth.t:1252]: @Firstsawyou 
Can we omit the `user-agent`? Since this field will change across different versions.

Comment (nic-chen) [reply t/plugin/hmac-auth.t]: > When requesting the URL of `/uri`, all request headers will be output. So it is normal to have the request header `user-agent`.

I mean it is not sorted in order.

Comment (Firstsawyou) [reply t/plugin/hmac-auth.t]: > > When requesting the URL of `/uri`, all request headers will be output. So it is normal to have the request header `user-agent`.
> 
> I mean it is not sorted in order.

It has been sorted, `user-agent` is after `host`.  ^ _ ^

**Review by spacewander — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

**Pull request merged**